samples/headless-ssr/commerce-generic/src/common/engine.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  type CommerceEngineDefinitionOptions,
+  defineCommerceEngine,
+  defineParameterManager,
+  defineProductList,
+  defineSearchBox,
+  defineSummary,
+  getSampleCommerceEngineConfiguration,
+} from '@coveo/headless/ssr-commerce-next';
+
+const config: CommerceEngineDefinitionOptions = {
+  configuration: getSampleCommerceEngineConfiguration(),
+  controllers: {
+    productList: defineProductList(),
+    summary: defineSummary(),
+    searchBox: defineSearchBox(),
+    parameterManager: defineParameterManager(),
+  },
+};
+
+export const engineDefinition = defineCommerceEngine(config);
+
+export type EngineDefinition = typeof engineDefinition.searchEngineDefinition;
```

Comment:
Maybe in a different PR, we would want to demo a listing, search and recommendation